File: Arrays _101/448. Find All Numbers Disappeared in an Array/findall.cpp
class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int> &nums)
    {
        vector<bool> ids;
        vector<int> non_existing;

        for (int i = 0; i <= nums.size(); i++)
        {
            ids.push_back(false);
        }

        for (int i = 0; i < nums.size(); i++)
        {
            ids[nums[i]] = true;
        }

        for (int i = 1; i <= nums.size(); i++)
        {
            if (!ids[i])
            {
                non_existing.push_back(i);
            }
        }
        return non_existing;
    }
};

File: 1. Arrays 101/448. Find All Numbers Disappeared in an Array/findall.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        ids = []
        non_existing = []

        for i in range(len(nums) + 1):
            ids.append(False)

        for i in range(len(nums)):
            ids[nums[i]] = True

        for i in range(1, len(nums) + 1):
            if not ids[i]:
                non_existing.append(i)

        return non_existing